src/core/orderbook.js

```diff
@@ -933,41 +933,44 @@ OrderBook.prototype.insertOffer = function(node) {
     log.info('inserting offer', this._key, node.fields);
   }
 
-  const offer = OrderBook.offerRewrite(node.fields);
-  const takerGets = this.normalizeAmount(this._currencyGets, offer.TakerGets);
-  const takerPays = this.normalizeAmount(this._currencyPays, offer.TakerPays);
+  try {
+    const offer = OrderBook.offerRewrite(node.fields);
+    const takerGets = this.normalizeAmount(this._currencyGets, offer.TakerGets);
+    const takerPays = this.normalizeAmount(this._currencyPays, offer.TakerPays);
 
-  offer.LedgerEntryType = node.entryType;
-  offer.index = node.ledgerIndex;
+    offer.LedgerEntryType = node.entryType;
+    offer.index = node.ledgerIndex;
 
-  // We're safe to calculate quality for newly created offers
-  offer.quality = takerPays.divide(takerGets).to_text();
+    // We're safe to calculate quality for newly created offers
+    offer.quality = takerPays.divide(takerGets).to_text();
```

Comment:
Outside of scope of this pull request, but we could look at the BookDirectory for the quality too